core/vtk/ttkDistanceMatrixDistorsion/ttkDistanceMatrixDistorsion.cpp

```diff
@@ -0,0 +1,151 @@
+#include <ttkDistanceMatrixDistorsion.h>
+
+#include <vtkInformation.h>
+
+#include <vtkDataArray.h>
+#include <vtkDataSet.h>
+#include <vtkDoubleArray.h>
+#include <vtkObjectFactory.h>
+#include <vtkPointData.h>
+#include <vtkSmartPointer.h>
+#include <vtkTable.h>
+#include <vtkVariantArray.h>
+
+#include <regex>
+
+#include <ttkMacros.h>
+#include <ttkUtils.h>
+
+// A VTK macro that enables the instantiation of this class via ::New()
+// You do not have to modify this
+vtkStandardNewMacro(ttkDistanceMatrixDistorsion);
+
+ttkDistanceMatrixDistorsion::ttkDistanceMatrixDistorsion() {
+  this->SetNumberOfInputPorts(2);
+  this->SetNumberOfOutputPorts(1);
+}
+
+int ttkDistanceMatrixDistorsion::FillInputPortInformation(
+  int port, vtkInformation *info) {
+  if(port == 0) {
+    info->Set(vtkAlgorithm::INPUT_REQUIRED_DATA_TYPE(), "vtkTable");
+    return 1;
+  } else if(port == 1) {
+    info->Set(vtkAlgorithm::INPUT_REQUIRED_DATA_TYPE(), "vtkTable");
+    return 1;
+  }
+  return 0;
+}
+
+int ttkDistanceMatrixDistorsion::FillOutputPortInformation(
+  int port, vtkInformation *info) {
+  if(port == 0) {
+    info->Set(vtkDataObject::DATA_TYPE_NAME(), "vtkTable");
+    return 1;
+  }
+  return 0;
+}
+
+inline void fillWithInputColumns(vtkTable *input,
+                                 const std::string &regExpStr,
+                                 std::vector<std::string> &vectToFill) {
+  // Select all input columns whose name is matching the regexp.
+  vectToFill.clear();
+  const size_t n = input->GetNumberOfColumns();
+  for(size_t i = 0; i < n; ++i) {
+    const auto &name = input->GetColumnName(i);
+    if(std::regex_match(name, std::regex(regExpStr))) {
+      vectToFill.emplace_back(name);
+    }
+  }
+}
+
+int ttkDistanceMatrixDistorsion::RequestData(
+  vtkInformation *ttkNotUsed(request),
+  vtkInformationVector **inputVector,
+  vtkInformationVector *outputVector) {
+
+  // Get input objects from input vector
+  vtkTable *inputHigh = vtkTable::GetData(inputVector[0]);
+  vtkTable *inputLow = vtkTable::GetData(inputVector[1]);
+
+  vtkTable *output = vtkTable::GetData(outputVector);
+
+  if(!inputLow || !inputHigh || !output)
+    return 0;
+
+  // Selecting the columns names if a regexp was given.
+  if(SelectFieldsWithRegexpHigh)
+    fillWithInputColumns(inputHigh, RegexpStringHigh, ScalarFieldsHigh);
+  if(SelectFieldsWithRegexpLow)
+    fillWithInputColumns(inputLow, RegexpStringLow, ScalarFieldsLow);
+
+  const size_t nRowsHigh = inputHigh->GetNumberOfRows();
+
+#ifndef TTK_ENABLE_KAMIKAZE
```

Comment:
I would remove the safeguard `TTK_ENABLE_KAMIKAZE` has this part of the code is not low-level (it is not continuously called by a calling program) and informing the user about a potentially invalid input is useful.